### ISSUE TITLE
[Bug] gather: RmSingleRowMultiCore returns wrong elements for row-major index rows wider than 1920

### ISSUE THREAD

**Issue by mmoscickiTT** [tenstorrent/tt-metal#55450]
### Summary

`ttnn.gather` returns wrong elements for a ROW_MAJOR index tensor whose last dim is wider than
`GATHER_WT_THRESHOLD * TILE_WIDTH` = `60 * 32` = **1920**. Above that width
`select_program_factory` switches from `RmSingleRowSingleCore` to `RmSingleRowMultiCore`
([`gather_device_operation.cpp`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/data_movement/gather/device/gather_device_operation.cpp#L17)),
and the multi-core factory produces incorrect output — roughly half the elements. No assertion
fires; the op returns silently wrong data.

This is pre-existing on `main` and unrelated to any codegen work. It surfaced because
tt-metal#52074 adds the first test in the repo that walks a ROW_MAJOR gather wider than 1920
(`tests/ttnn/nightly/unit_tests/operations/data_movement/test_gather_codegen_routing.py`), which
fails on `wh_n150_civ2`.

### Repro

```python
import torch, ttnn

dev = ttnn.open_device(device_id=0)
torch.manual_seed(0)
for N in (151936, 4096, 2048, 1024, 512, 64):
    x = torch.rand([1, N], dtype=torch.bfloat16)
    idx = torch.randint(0, N, [1, N], dtype=torch.int32)
    xt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT, device=dev)
    it = ttnn.from_torch(idx, dtype=ttnn.uint32, layout=ttnn.ROW_MAJOR_LAYOUT, device=dev)
    a = ttnn.to_torch(ttnn.gather(xt, dim=-1, index=it)).contiguous().flatten()
    ref = torch.gather(x, -1, idx.long()).contiguous().flatten()
    print(f"N={N:7d} mismatch={(a != ref).sum().item():7d}")
ttnn.close_device(dev)
```

### Measured (Blackhole p150b, `main`, no codegen branch)

| index width `N` | factory selected | mismatches vs `torch.gather` |
|---|---|---|
| 64 | `RmSingleRowSingleCore` | 0 |
| 512 | `RmSingleRowSingleCore` | 0 |
| 1024 | `RmSingleRowSingleCore` | 0 |
| 2048 | `RmSingleRowMultiCore` | **1199 / 2048** |
| 4096 | `RmSingleRowMultiCore` | **2190 / 4096** |
| 151936 | `RmSingleRowMultiCore` | **74468 / 151936** |

The break lands exactly on the 1920 boundary: every width below it is clean, every width above it
is wrong.

### Not an index-dtype problem

An early hypothesis was index truncation to 16 bits. Ruled out: a `uint16` index at `N = 2048`
gives **1610** mismatches — *worse* than `uint32`'s 1199 at the same width. The failure tracks the
factory choice, not the index dtype.

### Why it went unnoticed

`main` has no ROW_MAJOR gather coverage at all — the existing gather tests are TILE layout, where
a different factory runs. The wrong elements are usually plausible finite bfloat16 values, so a
PCC-style check on a narrow shape would not have caught it either; on some SKUs the garbage decodes
to `NaN`, which is the only reason the new test fails loudly on `wh_n150_civ2` and passes on
`bh_p150b_civ2` / `wh_n300_civ2`.

### Impact

Any caller doing a row-major gather with an index row wider than 1920 gets silently wrong results.
That shape is not exotic — any vocabulary-sized gather hits it directly.

### Meanwhile

tt-metal#52074 narrows its wide row-major case to assert the routing property only, so the port PR
is not blocked on this fix. That test should be restored to a full value comparison once this is
fixed.
